### PR TITLE
503 shinytest2 for `teal_data_module`

### PR DESCRIPTION
**Pull request by averissimo**
# Pull Request

Part of https://github.com/insightsengineering/coredev-tasks/issues/503

#### Changes description

- Adds tests for `teal_data_module`
    1. Loads `iris` and `mtcars`
    2. Fails to load `teal` app if required input is not available
    3. Integrates input on `teal_data()` object by creating a new dynamic column

### CODE REVIEW

File: tests/testthat/test-shinytest2-teal_data_module.R [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  )
+
+  app$click("teal_data_module-submit")
+  app$wait_for_idle()
+  testthat::expect_setequal(app$get_active_filter_vars(), c("dataset1", "dataset2"))
+
+  app$stop()
+})
+
+testthat::test_that("e2e: teal_data_module fails to proceed without input", {
```
Comment (vedhav) [inline tests/testthat/test-shinytest2-teal_data_module.R:38]: I like the test. However, I think the desc can be renamed. I believe that we are testing "tdm shows any validation errors" else we're just testing this specific instance of the app.

Comment (vedhav) [reply tests/testthat/test-shinytest2-teal_data_module.R:38]: There is a `app$expect_validation_error()` to test this.

File: tests/testthat/test-shinytest2-teal_data_module.R [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  app$wait_for_idle()
+
+  app$click("teal_data_module-submit")
+
+  app$expect_validation_error()
+
+  app$stop()
+})
+
+testthat::test_that("e2e: teal_data_module adds new column to datasets", {
```
[the remark this answers is not in the document]
Comment (averissimo) [reply tests/testthat/test-shinytest2-teal_data_module.R:75]:
```suggestion
testthat::test_that("e2e: teal_data_module inputs change data that is passed to teal main UI", {
```

File: tests/testthat/test-shinytest2-teal_data_module.R (hunk 1 above)
Comment (averissimo) [reply tests/testthat/test-shinytest2-teal_data_module.R]: :+1: Renamed it to your suggestion 016d28f

`expect_validation_error` is being used, what do you mean?

Comment (vedhav) [reply tests/testthat/test-shinytest2-teal_data_module.R]: True, Sorry I did not see the `expect_validation_error`.

**Review by vedhav — APPROVED** [review]
LGTM!

**Pull request merged**